lib/plugins/aws/deploy/compile/events/apiGateway/lib/methods.js

```diff
@@ -6,6 +6,10 @@ const _ = require('lodash');
 module.exports = {
   compileMethods() {
     _.forEach(this.serverless.service.functions, (functionObject, functionName) => {
+      if (!functionObject.events) {
```

Comment:
The idea was to keep the diff as small as possible, but I agree with you. I have amended the commit.